### ISSUE TITLE
Add `uv init --from-project <path>` to import from other tools

### ISSUE THREAD

**Issue by zanieb** [astral-sh/uv#6277]
As previously discussed for `uv add -r pyproject.toml` but without any of the controversy about what to do with extras. We can copy all of the dependencies as reported by the build backend.

**Comment by charliermarsh** [comment]
I'll look into this tonight unless someone beats me to it.

**Comment by charliermarsh** [comment]
I have no idea why I closed this.

**Comment by charliermarsh** [comment]
Related: https://github.com/astral-sh/uv/issues/6164

**Comment by blueraft** [comment]
Is this up for grabs?

**Comment by charliermarsh** [comment]
Yes!

**Comment by charliermarsh** [comment]
You can use `SourceTreeResolver` to get the metadata. But one challenge is... we probably need to preserve extras rather than resolving them. For example, if the user's `pyproject.toml` has:

```
[project.optional-dependencies]
foo = "bar"
```

Then in the project we create, we'd _also_ want to create a `foo` extra with `bar` in it.

**Comment by blueraft** [comment]
I'll take a look soon. 

> we'd also want to create a foo extra with bar in it.

This makes sense, a similar PR would then resolve #6164 right?

**Comment by charliermarsh** [comment]
Yeah, it's possible that we want to skip #6164 for now since it brings up some confusing questions, like what does `--extra` mean in `uv add --extra baz -r pyproject.toml`. So let's start with support here in `uv init`.

**Comment by mrijken** [comment]
What is needed to get this merged/released? It will help converting a lot of users  from Poetry to UV, at least for companies which have large number of repo's.

**Comment by mrcljx** [comment]
Shameless plug (in case someone stumbles over this issue here and `--from-project` is still WIP): I wrote a script to migrate our Poetry monorepo project to `uv`. I doubt it will work immediately on other projects, but could be a good start.

https://gist.github.com/mrcljx/8e99a7f9fb924a573416f4365ee15873

**Comment by Defiect** [comment]
Any update on this?
